src/components/MyAccount/Settings/SettingsInputForm.tsx

```diff
@@ -183,10 +208,15 @@ const SettingsInputForm = ({
                     sx={{
                       width: { base: "87%", md: "300px" },
                     }}
+                    ref={(el) => (inputRefs.current[index] = el)}
```

Comment:
Not sure if you came across [this](https://react.dev/learn/manipulating-the-dom-with-refs), but your solution is react.dev approved! Looking at their solution, though, I think it might be important to remember to delete refs from the array once the input is removed from the dom.

Reply:
I tried adding 5 inputs, then deleting one from the middle, and the count on the array of input refs actually went up to 6. It doesn't seem to cause any problems, but I think to avoid bugs down the line, the inputRefs array should match the inputs that are actually present on the dom.

Reply:
Actually just encountered a bug! after opting to stay logged in from the timed logout modal, the focus is lost and the last input is null, which leads to a loss of focus. I think  this can be avoided by removing refs as the inputs are removed.